src/common/Header/SearchPanel/index.js

```diff
@@ -1,14 +1,42 @@
-import React from "react";
-import { useLocation } from "react-router-dom";
+import React, { useEffect, useRef } from "react";
+import { useHistory, useLocation } from "react-router-dom";
 import { toMovies } from "../../../routes";
 import { SearchInput } from "./styled";
+import { useQueryParameter } from "../../../useQueryParameters";
+import { search as searchParameterName } from "../../../queryParamNames";
 
 const SearchPanel = () => {
     const location = useLocation();
+    let query = useQueryParameter(searchParameterName);
+    let timeout = 0;
+    const history = useHistory();
+    const inputRef = useRef(null);
+
+    useEffect(() => {
+        inputRef.current.value = query;
+    }, [location]);
+
+    const onInputChange = ({ target }) => {
+        if (timeout) {
+            clearTimeout(timeout);
+        }
+        timeout = setTimeout(() => {
+            const path = location.pathname.includes(toMovies()) ? "movies" : "people";
+            query = target.value;
+            if (query) {
+                history.push(`/${path}?${searchParameterName}=${query}`)
+            }
+            else {
+                history.push(`/`)
```

Comment:
a tutaj zamiast history.push(`/`) zrobiłbym history.push(`/${pathname}`), gdyż jeśli backspace'm będziemy czyścić input search w podsekcji people i wyczyśnimy go to przekierowuje nas na movies, a tego byśmy nie chcieli ;)

Reply:
a dokładniej to `/{path}`
:) i tak, należy to wprowadzić jak najbardziej

Reply:
Oczywiście @TSypko  😃  za dużo dopisałem 🤣